File: web/src/api/toolConfig.ts
import request from '@/utils/request'

export interface ToolConfig {
  id: number
  tool_code: string
  tool_name: string
  enabled: boolean
  visible: boolean
  is_deprecated: boolean
  config_schema: string
  description: string
  category: string
  version: string
  author: string
  tags: string
  sort_order: number
  last_sync_at: string
  created_at: string
  updated_at: string
}

export interface ToolConfigDetail extends ToolConfig {
  decrypted_config: Record<string, any>
}

export interface UpdateToolConfigRequest {
  config: Record<string, any>
}

export interface UpdateToolSettingsRequest {
  enabled: boolean
  visible: boolean
  sort_order: number
}

export interface ToolCategory {
  code: string
  name: string
  description: string
  icon: string
}

/**
 * 获取所有工具配置（管理端）
 */
export function getAllToolConfigs() {
  return request<ToolConfig[]>({
    url: '/api/v1/admin/tool-configs',
    method: 'GET',
  })
}

/**
 * 获取工具配置详情（含解密后的配置）
 */
export function getToolConfigDetail(toolCode: string) {
  return request<ToolConfigDetail>({
    url: `/api/v1/admin/tool-configs/${toolCode}`,
    method: 'GET',
  })
}

/**
 * 更新工具配置
 */
export function updateToolConfig(toolCode: string, data: UpdateToolConfigRequest) {
  return request({
    url: `/api/v1/admin/tool-configs/${toolCode}`,
    method: 'PUT',
    data,
  })
}

/**
 * 更新工具设置（启用/禁用、可见性、排序）
 */
export function updateToolSettings(toolCode: string, data: UpdateToolSettingsRequest) {
  return request({
    url: `/api/v1/admin/tool-configs/${toolCode}/settings`,
    method: 'PATCH',
    data,
  })
}

/**
 * 删除工具配置
 */
export function deleteToolConfig(id: number) {
  return request({
    url: `/api/v1/admin/tool-configs/${id}`,
    method: 'DELETE',
  })
}

/**
 * 同步工具定义
 */
export function syncTools() {
  return request({
    url: '/api/v1/admin/tool-configs/sync',
    method: 'POST',
  })
}

/**
 * 获取工具分类列表
 */
export function getToolCategories() {
  return request<ToolCategory[]>({
    url: '/api/v1/tool-categories',
    method: 'GET',
  })
}

File: internal/models/tool_config.go
package models

import (
	"time"
)

// ToolConfig 工具配置模型
type ToolConfig struct {
	ID           uint      `gorm:"primaryKey" json:"id"`
	ToolCode     string    `gorm:"type:varchar(50);uniqueIndex;not null;comment:工具代码" json:"tool_code"`
	ToolName     string    `gorm:"type:varchar(100);not null;comment:工具名称" json:"tool_name"`
	Enabled      bool      `gorm:"default:true;comment:是否启用" json:"enabled"`
	Visible      bool      `gorm:"default:true;comment:是否对外展示" json:"visible"`
	IsDeprecated bool      `gorm:"default:false;comment:是否已废弃" json:"is_deprecated"`
	ConfigJSON   string    `gorm:"type:text;comment:配置JSON(加密)" json:"-"` // 不返回给前端
	ConfigSchema string    `gorm:"type:text;comment:配置结构定义JSON" json:"config_schema"`
	Description  string    `gorm:"type:varchar(500);comment:工具描述" json:"description"`
	Category     string    `gorm:"type:varchar(50);comment:工具分类" json:"category"`
	Version      string    `gorm:"type:varchar(20);comment:工具版本" json:"version"`
	Author       string    `gorm:"type:varchar(100);comment:工具作者" json:"author"`
	Tags         string    `gorm:"type:varchar(500);comment:工具标签(逗号分隔)" json:"tags"`
	SortOrder    int       `gorm:"default:0;comment:排序" json:"sort_order"`
	LastSyncAt   time.Time `gorm:"comment:最后同步时间" json:"last_sync_at"`
	CreatedAt    time.Time `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt    time.Time `gorm:"autoUpdateTime" json:"updated_at"`
}

// TableName 指定表名
func (ToolConfig) TableName() string {
	return "tool_config"
}
